packages/mobile/src/services/http/http.service.ts

```diff
@@ -33,14 +41,43 @@ class Http {
       hasAuth,
     });
 
-    return fetch(this.getUrl(url, params), {
+    return this.makeRequest(this.getUrl(url, params), {
       method,
       headers,
       body: payload,
-    })
-      .then(this.checkStatus)
-      .then((res) => this.parseJSON<T>(res))
-      .catch(this.throwError);
+    });
+  }
+
+  private async makeRequest<T = unknown>(
+    url: string,
+    config: { method: HttpMethod; headers: Headers; body: BodyInit_ },
+  ): Promise<T> {
+    const result = await fetch(url, config);
+
+    if (result.status === HttpStatusCode.UNAUTHORIZED) {
+      await this.updateAuthorizationToken();
+      const updatedConfig = this.getRefreshedAuthReqConfig(config);
```

Comment:
This logic seems to belong to `checkStatus`

Reply:
it makes sense. but it seems to me that in this case it is not very convenient to pass the config between the initial and the next fetch after updating the token